packages/ui/src/components/va-hover/VaHover.vue

```diff
@@ -40,3 +40,10 @@ export default defineComponent({
   },
 })
 </script>
+
+<style lang="scss">
+.va-hover {
+  dispaly: flex;
```

Comment:
Typo here.

```suggestion
  display: flex;
```

Reply:
I think you need to use `flex-grow` or `width: 100%` class in your example, if you want VaHover to fit your parent component.

For the most common example hover must flow its content:
```vue
<va-hover>
   <va-button>text</va-button>
</va-hover>
```

In current implementation, there is a bug, so `va-hover` is not flow `va-button` and hover is triggered outside of content.
![firefox_T46ksVG3NC](https://user-images.githubusercontent.com/23530004/198414853-29ae07e5-34b5-431e-b57b-676fa7a77f00.gif)

Reply:
So I would prefer fixing current:
![image](https://user-images.githubusercontent.com/23530004/198416305-45db7090-5b27-4913-bc2f-480ef883bd04.png)

to:
![image](https://user-images.githubusercontent.com/23530004/198416358-651b1108-759b-410a-bef9-303435df71bc.png)

Which makes more sense.

Also, setting width: 100% to hover container child must fill VaHover
![image](https://user-images.githubusercontent.com/23530004/198416587-cc5330d3-ea52-4762-b584-b2f3d866656d.png)

So it behaves more like headless component.

Reply:
So my suggestion will be
```suggestion
  display: inline-flex;
  & > * {
     flex: 1;
  }
```

Reply:
VaHover, because there is an additional layer of dom structure, I think It's a doomed road 
 whether it is `dispaly:inline-flex`, or `dispaly:flex;flex:1`,
Neither is a good way, if you have to choose between the two, 
choose `dispaly:inline-flex`

Finally, I think there should be a non-component way to use hover state, like
1. Instructions
2. useElementHover

Considering implementing a v-directive? 
 you can  look at https://vueuse.org/core/useelementhover/

Reply:
@Valar103769, agree with that.
I would like to see a headless component instead of current implementation. Until then, I would recommend using vueuse composable.